rclpy/rclpy/utilities.py

```diff
@@ -39,6 +39,13 @@ def get_default_context(*, shutting_down=False):
         return g_default_context
 
 
+def get_domain_id_in_context(context=None):
```

Comment:
This utility function doesn't sound necessary.

Doing `get_default_context().get_domain_id()` or `my_context.get_domain_id()` look better than passing the context to a function.

Reply:
@ivanpauno 

This utility function is only for context' get_domain_id(). 
```python
class Context:
    ...
    def get_domain_id(self):
    """Get domain id of context."""
    from rclpy.impl.implementation_singleton import rclpy_implementation
    with self._handle as capsule, self._lock:
         return rclpy_implementation.rclpy_context_get_domain_id(capsule)
```

This way likes context' ok().  
```python
class Context:
    ...
    def ok(self):
        """Check if context hasn't been shut down."""
        # imported locally to avoid loading extensions on module import
        from rclpy.impl.implementation_singleton import rclpy_implementation
        with self._handle as capsule, self._lock:
            return rclpy_implementation.rclpy_ok(capsule)
```

Reply:
Yeah, but we don't want to have all the methods of `Context` as free functions.

In the case of `ok`, we use that in every spinning loop. So having a free function was a good idea:

```python3
try:
    while rclpy.ok():
        ...
        rclpy.spin_once()
        ...
except KeyboardInterrupt:
    ...
```

In this case, a free function sounds unnecessary.